src/treemodel/treewalker.js

```diff
@@ -44,6 +44,18 @@ export default class TreeWalker {
 			throw new CKEditorError( 'tree-walker-no-start-position: Neither boundaries nor starting position have been defined.' );
 		}
 
+		if ( options && options.direction && [ 'FORWARD', 'BACKWARD' ].indexOf( options.direction ) < 0 ) {
```

Comment:
In fact we could use destucturing here:
```
constructor( { 
	boundaries,
	startPosition,
	singleCharacters=false,
	shallow=false,
	ignoreElementEnd=false,
	direction == 'FORWARD'
 } = {} )
```
Then use variables directly in code (`direction` instead of `options.direction` ).